app/components/questionnaire/answer-form-list.js

```diff
@@ -48,30 +46,17 @@ const AnswerFormList = Ember.Component.extend({
       return JSON.stringify(each.cwlType) === JSON.stringify((cwlType));
     });
   },
+
   actions: {
-    /*
-     provideAnswer is passed down to the individual component.
-     when user answers the question in the component, it is called
-    * */
-    provideAnswer(answer) {
+    answerChanged(answerComponent) {
+      // Answer components will send this action when their answer changes
+      // When that happens, update the answer and any input files
       const answerSet = this.get('answerSet');
-      // Promote to Ember.Object so that we can call setProperties
       let userJobOrder = Ember.Object.create(answerSet.get('userJobOrderJson'));
-      userJobOrder.setProperties(answer);
+      userJobOrder.setProperties(answerComponent.get('answer'));
       answerSet.set('userJobOrderJson', userJobOrder);
-      return Ember.RSVP.resolve();
-    },
-    provideInputFiles(inputFiles) {
-      return this.get('stageGroup')
-        .then(stageGroup => { return stageGroup.save(); })
-        .then(stageGroup => {
-          return Ember.RSVP.all(
-            inputFiles.map(inputFile => {
-              inputFile.set('stageGroup', stageGroup);
-              return inputFile.save();
-            })
-          );
-        });
+      // set the stage group in the files
+      answerComponent.get('inputFiles').setEach('stageGroup', answerSet.get('stageGroup'));
```

Comment:
> specific to the particular answer components?

Yes and no. It's needed by the one answer component we have (`file-group-list`), but will apply to any answer component that supplies files needing staging.

My thinking is that answerComponents generically will supply two properties for the answer list to consume:

- `answer`:  an object with its part of the CWL JSON
- `inputFiles`: an array of `dds-job-input-file`s or `url-job-input-file`s. Could be an empty array or undefined.

If a component doesn't need to supply any files (e.g. just a string or a number) it doesn't have to supply anything in the array. Probably best to check if this is present first.

I don't think we'll need a different line here in `answer-form-list` for every type of component.